### PR TITLE
test(e2e): add homepage 

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem
Previously, homepage did not have any e2e tests. This PR adds e2e tests for homepage.

## Solution
1. change dnd package - old dnd package had an issue with react strict mode, which causes it to crash on local.
2. add tests for special cases and drag/drop functionality
    - see the base drag and drop spec from react beautiful dnd [here](https://github.com/atlassian/react-beautiful-dnd/blob/master/cypress/integration/reorder.spec.js) 
    - other special cases are given on the ticket and guarntee some baseline behaviour

### CODE REVIEW

File: cypress/e2e/homepage.spec.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+const REACT_DRAG_HANDLE_PROPERTY = "data-rfd-drag-handle-draggable-id" as const
+
+const getHandleSelector = (draggableId?: string) => {
+  if (draggableId) {
+    return `[${REACT_DRAG_HANDLE_PROPERTY}="${draggableId}"]`
+  }
+  return `[${REACT_DRAG_HANDLE_PROPERTY}]`
+}
```
Comment (alexanderleegs) [inline cypress/e2e/homepage.spec.ts:61]: Could we consider moving this selector (and the associated reordering) into a utils file? Nav bar/Contact us also make use of this and are similarly missing tests

**Review by alexanderleegs — COMMENTED** [review]
I think it might also be worth testing individual component behaviour (specifically for infopic, which contains uploading of images from the modal), as well as attempting to save the overall page to ensure that changes don't mess up the expected formatting!

File: cypress/e2e/homepage.spec.ts (hunk 1 above)
Comment (seaerchin) [reply cypress/e2e/homepage.spec.ts:61]: can! i just left it here first because the other tests aren't up

**Comment by kishore03109** [comment]
!run-chromatic

File: cypress/e2e/homepage.spec.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      // Act
+      // reorder operation
+      cy.get("@first")
+        .focus()
+        .trigger("keydown", { keyCode: keyCodes.space })
+        // need to re-query for a clone
+        .get("@first")
+        .trigger("keydown", { keyCode: keyCodes.arrowDown, force: true })
+        // finishing before the movement time is fine - but this looks nice
+        .wait(timings.outOfTheWay * 1000)
```
Comment (QiluXie) [inline cypress/e2e/homepage.spec.ts:181]: Just curious, how did we come up with the time that we need to wait here (0.2s)?

File: cypress/fixtures/constants.ts [hunk 3]
```diff
@@ -61,3 +61,17 @@ export const MOCK_REVIEW_DESCRIPTION = "Some interesting description"
...
 export const FORBIDDEN_CHARACTERS = "!@#$%^&*()"
 export const NON_ENGLISH_CHARACTERS = "文கி"
+
+export const timings = {
+  outOfTheWay: 0.2,
+  // greater than the out of the way time
+  // so that when the drop ends everything will
+  // have to be out of the way
+  minDropTime: 0.33,
+  maxDropTime: 0.55,
```
Comment (QiluXie) [inline cypress/fixtures/constants.ts:71]: Couldn't find usage of these two constants, will they be used in future test? 
Same question, how did we come up with the values here?

File: cypress/e2e/homepage.spec.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+      cy.get(ADD_SECTION_DROPDOWN_SELECTOR).select(
+        SECTION_DROPDOWN_OPTIONS.RESOURCES
+      )
+      cy.get("@addSectionButton").click()
+
+      // Assert
+      // NOTE: This is an existing bug on `production`
+      // where adding a resource section doesn't disable.
+      cy.get("@addSectionButton").should("be.disabled")
+    })
```
Comment (QiluXie) [inline cypress/e2e/homepage.spec.ts:148]: It means this assertion will fail if we don't fix the existing bug in production, right?

File: cypress/e2e/homepage.spec.ts [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+        // need to re-query for a clone
+        .get("@first")
+        .trigger("keydown", { keyCode: keyCodes.arrowDown, force: true })
+        // finishing before the movement time is fine - but this looks nice
+        .wait(timings.outOfTheWay * 1000)
+        .trigger("keydown", { keyCode: keyCodes.space, force: true })
+
+      // Assert
+      // NOTE: Order is now Infopic -> Infobar
+      // note: not using get aliases as they where returning incorrect results
```
Comment (QiluXie) [inline cypress/e2e/homepage.spec.ts:186]: Super nit: typo in the comment? should be `will` instead of `where`?

**Review by QiluXie — COMMENTED** [review]
Looks good, left some minor comments (mostly questions).

File: cypress/e2e/homepage.spec.ts (hunk 2 above)
Comment (seaerchin) [reply cypress/e2e/homepage.spec.ts:181]: it's the time given by the authors of the library in their tests (see [here](https://github.com/atlassian/react-beautiful-dnd/blob/master/cypress/integration/reorder.spec.js)).

File: cypress/fixtures/constants.ts (hunk 3 above)
Comment (seaerchin) [reply cypress/fixtures/constants.ts:71]: read the test spec for `react-beautiful-dnd` (see [here](https://github.com/atlassian/react-beautiful-dnd/blob/master/cypress/integration/reorder.spec.js)) and took these values

File: cypress/e2e/homepage.spec.ts (hunk 4 above)
Comment (seaerchin) [reply cypress/e2e/homepage.spec.ts:148]: yeap, that's correct.

File: cypress/e2e/homepage.spec.ts (hunk 5 above)
Comment (seaerchin) [reply cypress/e2e/homepage.spec.ts:186]: typo from the lib spec itself AHHAHA will fix

File: cypress/e2e/homepage.spec.ts [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+      // Act
+      cy.get(ADD_SECTION_DROPDOWN_SELECTOR).select(
+        SECTION_DROPDOWN_OPTIONS.RESOURCES
+      )
+      cy.get("@addSectionButton").click()
+
+      // Assert
+      // NOTE: This is an existing bug on `production`
+      // where adding a resource section doesn't disable.
```
Comment (harishv7) [inline cypress/e2e/homepage.spec.ts:146]: are we planning to fix this? is there a ticket?

File: cypress/e2e/homepage.spec.ts [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+      cy.get("@addSectionButton").should("be.disabled")
+    })
+
+    it.skip("should snap preview to that section when clicking on the section in the editor", () => {
+      // NOTE: Not testing this because cypress doesn't support viewports checking
+    })
+
+    it("should rearrange the blocks correctly and the preview should also reflect the new order", () => {
```
Comment (harishv7) [inline cypress/e2e/homepage.spec.ts:154]: Screenshot 2023-08-02 at 10 37 28 AM

Somehow I can't get the last test to pass, it keeps failing after the click

Comment (seaerchin) [reply cypress/e2e/homepage.spec.ts:154]: Oh I think here it's not react beautiful DND so the selector id might be different

File: cypress/e2e/homepage.spec.ts (hunk 6 above)
Comment (seaerchin) [reply cypress/e2e/homepage.spec.ts:146]: Unsure, I think it's good to fix but it's difficult due to how messy this section of the code is...

Comment (harishv7) [reply cypress/e2e/homepage.spec.ts:146]: Hmm, then the test will keep failing right? Shall we comment this assert with a TODO on top?

ELse, it becomes like a legacy knowledge that oncalls need to know this failure is expected

File: cypress/e2e/homepage.spec.ts (hunk 7 above)
Comment (harishv7) [reply cypress/e2e/homepage.spec.ts:154]: hmm so do we need to update the selector?

File: cypress/e2e/homepage.spec.ts (hunk 6 above)
Comment (seaerchin) [reply cypress/e2e/homepage.spec.ts:146]: Sure, sounds food

File: cypress/e2e/homepage.spec.ts (hunk 7 above)
Comment (seaerchin) [reply cypress/e2e/homepage.spec.ts:154]: Haven't try yet, will let you know.

Comment (harishv7) [reply cypress/e2e/homepage.spec.ts:156]: resolving as this was tested against staging (not local)

**Review by harishv7 — APPROVED** [review]

**Comment by seaerchin** [comment]
Screenshot 2023-08-02 at 3 32 42 PM
 
pic for proof

**Pull request merged**